src/zenml/models/v2/core/trigger.py

```diff
@@ -0,0 +1,133 @@
+#  permissions and limitations under the License.
+"""Collection of all models concerning triggers."""
+from datetime import datetime
+from typing import Any, Dict, Optional, Union
+from uuid import UUID
+
+from pydantic import BaseModel, Field
+
+from zenml.constants import STR_FIELD_MAX_LENGTH
+from zenml.models.v2.base.scoped import (
+    WorkspaceScopedFilter,
+    WorkspaceScopedRequest,
+    WorkspaceScopedResponse,
+    WorkspaceScopedResponseBody,
+    WorkspaceScopedResponseMetadata,
+)
+from zenml.models.v2.base.update import update_model
+
+# ------------------ Base Model ------------------
+
+
+class TriggerBase(BaseModel):
+    """Base model for triggers."""
+
+    name: str = Field(
+        title="The name of the Trigger.", max_length=STR_FIELD_MAX_LENGTH
+    )
+    description: str = Field(
+        default="",
+        title="The description of the trigger",
+        max_length=STR_FIELD_MAX_LENGTH,
+    )
+    event_source_id: UUID
+    event_filter: Dict[str, Any]
+    action_plan: Dict[str, Any]
+    action_plan_flavor: str
+
+
+# ------------------ Request Model ------------------
+class TriggerRequest(TriggerBase, WorkspaceScopedRequest):
+    """Model for creating a new Trigger."""
+
+    # executions: somehow we need to link to executed Actions here
+
+
+# ------------------ Update Model ------------------
+
+
+@update_model
+class TriggerUpdate(TriggerRequest):
+    """Update model for stacks."""
+
+    name: str = Field(
+        title="The name of the Trigger.", max_length=STR_FIELD_MAX_LENGTH
+    )
+    description: str = Field(
+        default="",
+        title="The description of the trigger",
+        max_length=STR_FIELD_MAX_LENGTH,
+    )
+    event_filter: Dict[str, Any]
+    action_plan: Dict[str, Any]
+
+
+# ------------------ Response Model ------------------
+
+
+class TriggerResponseBody(WorkspaceScopedResponseBody):
+    """ResponseBody for triggers."""
+
+    created: datetime = Field(
+        title="The timestamp when this trigger was created."
+    )
+    updated: datetime = Field(
+        title="The timestamp when this trigger was last updated.",
+    )
+    action_plan_flavor: str
+    event_flavor: str
+
+
+class TriggerResponseMetadata(WorkspaceScopedResponseMetadata):
+    """Response metadata for triggers."""
+
+    event_filter: Dict[str, Any] = Field(
+        title="The event that activates this trigger.",
+    )
+    action_plan: Dict[str, Any] = Field(
+        title="The actions that is executed by this trigger.",
+    )
+
+
+class TriggerResponse(
+    WorkspaceScopedResponse[TriggerResponseBody, TriggerResponseMetadata]
+):
+    """Response model for models."""
+
+    name: str = Field(
+        title="The name of the model",
+        max_length=STR_FIELD_MAX_LENGTH,
+    )
+    description: str = Field(
+        default="",
+        title="The description of the trigger",
+        max_length=STR_FIELD_MAX_LENGTH,
+    )
```

Comment:
same here: these should probably go in the body or metadata.

this is also missing the properties referencing fields in the body and metadata